*add rsync action

hjanuschka opened the pull request:
a little wrapper around rsync  - to not have to use `sh`

``` ruby
rsync(
          source: "root@dev.januschka.com:/tmp/1.txt",
          destination: "/tmp/new_f"
    )
```

hjanuschka committed:
*add rsync action

KrauseFx commented on lib/fastlane/actions/rsync.rb:
```diff
@@ -0,0 +1,106 @@
+    class RsyncAction < Action
+
+      def self.run(params)
+        rsync_cmd = "rsync "
```
A more elegant way is to collect an array:
```ruby
rsync_cmd = ["rsync"]
rsync_cmd << params[:extra]
...
Actions.sh(rsync_cmd.join(" "))
```
:+1:

KrauseFx commented on lib/fastlane/actions/rsync.rb:
```diff
@@ -0,0 +1,106 @@
+
+        # Below a few examples
+        [
+          FastlaneCore::ConfigItem.new(key: :username,
```
This variable is never used

KrauseFx commented:
Thanks for the PR. Could you please remove all comments that are not written by you? It seems like some variables are never used. 
Could you also add a basic test to this integration? You can see more about how to run the tests in [countdown](https://github.com/fastlane/countdown). Let me know if I can help you with this :+1: In general, check out `swiftlint_specs` on how to easily test command line integrations.

KrauseFx commented:
Also, this needs to be added to `Actions.md`

hjanuschka commented:
* removed default comments
 * added basic rspec test
 * rubocop is fine
 * command building like you suggested
 * sample in Actions.md

@KrauseFx

KrauseFx commented on docs/Actions.md:
```diff
@@ -1746,6 +1746,17 @@ It can process unit test results if formatted as junit report as shown in [xctes
 
+### rsync
+
+a wrapper around rsync.
```
Instead it would be nice to just describe what it's for (e.g. file transfer) so that someone who's not familiar with `rsync` yet knows what it's about

KrauseFx commented on docs/Actions.md:
````diff
@@ -1746,6 +1746,17 @@ It can process unit test results if formatted as junit report as shown in [xctes
+
+```ruby
+    rsync(
+          source: "root@host:/tmp/1.txt",
````
Wrong tabs here

KrauseFx commented on lib/fastlane/actions/rsync.rb:
```diff
@@ -0,0 +1,69 @@
+      #####################################################
+
+      def self.description
+        "a wrapper arround rsync"
```
Same as in `Actions.md`, describe what it does

KrauseFx commented on lib/fastlane/actions/rsync.rb:
```diff
@@ -0,0 +1,69 @@
+      end
+
+      def self.available_options
+        # Define all options your action supports.
```
Please remove those comments here

KrauseFx commented on lib/fastlane/actions/rsync.rb:
```diff
@@ -0,0 +1,69 @@
+        [
+          FastlaneCore::ConfigItem.new(key: :extra,
+                                       short_option: "-X",
+                                       env_name: "FL_RSYNC_EXTRA", # The name of the environment variable
```
Please remove those comments here

KrauseFx commented on lib/fastlane/actions/rsync.rb:
```diff
@@ -0,0 +1,69 @@
+      end
+
+      def self.authors
+        # So no one will ever forget your contribution to fastlane :) You are awesome btw!
```
Please remove those comments here

KrauseFx commented on spec/actions_specs/rsync_spec.rb:
```diff
@@ -0,0 +1,12 @@
+        result = Fastlane::FastFile.new.parse("lane :test do
+          rsync(source: '/tmp/1.txt', destination: '/tmp/2.txt')
+        end").runner.execute(:test)
+        expect(result).to eq("rsync -av /tmp/1.txt /tmp/2.txt")
```
:heart:

KrauseFx commented:
Thanks for updating the PR, this is great, however we have merge conflicts (I assume with your other PR), could you resolve that? 
Also I added some more comments before it's ready to be merged. Please squash the commits. Thanks again for your great contribution :+1:

hjanuschka commented:
* feedback implemented

KrauseFx commented on lib/fastlane/actions/rsync.rb:
```diff
@@ -0,0 +1,65 @@
+      end
+
+      def self.details
+        "a wrapper around rsync, rsync is a tool that lets you synchronize files, including permissions and so on for a more detailed information about rsync please see ***rsync(1)*** manpage."
```
Please start with `A`

KrauseFx replied:
Why the `***`

KrauseFx commented:
Looks good, however we now have a merge conflict again, sorry :cry:

hjanuschka commented:
* ok i removed the Actions.md from this branch - as if you merge this one - the other PR's conflict too
 * you can merge this branch, and later on take the right block from   (https://gist.github.com/hjanuschka/96cb3129fd0d839b7a56) - or take a look at my combined docs PR.

https://github.com/fastlane/fastlane/pull/1207

@KrauseFx

KrauseFx commented:
That's perfect, thank you so much @hjanuschka :+1:
